apps/wallet-mobile/src/features/Portfolio/common/hooks/useGetPortfolioTokenChart.ts

```diff
@@ -70,45 +78,49 @@ export const useGetPortfolioTokenChart = (
   const {currency} = useCurrencyPairing()
   const {languageCode} = useLanguage()
 
-  const ptQuery = useQuery({
+  const ptPriceQuery = useQuery<
+    Awaited<ReturnType<typeof ptPriceQueryFn>>,
+    Error,
+    Awaited<ReturnType<typeof ptPriceQueryFn>>,
+    ['ptPriceHistory', TokenChartInterval]
+  >({
+    enabled: tokenInfo && isPrimaryToken(tokenInfo.info),
     staleTime: time.halfHour,
     cacheTime: time.oneHour,
-    retryDelay: time.oneSecond,
-    optimisticResults: true,
     refetchInterval: time.halfHour,
-    useErrorBoundary: true,
-    refetchOnMount: false,
+    queryKey: ['ptPriceHistory', timeInterval],
+    queryFn: ptPriceQueryFn,
+  })
+
+  const ptQuery = useQuery({
     enabled: tokenInfo && isPrimaryToken(tokenInfo.info),
+    staleTime: time.oneMinute,
     ...options,
-    queryKey: ['useGetPortfolioTokenChart', tokenInfo?.info.id ?? '', timeInterval, currency],
+    queryKey: ['useGetPortfolioTokenChart', 'pt', timeInterval, currency],
     queryFn: async () => {
-      const response = await fetchPtPriceActivity(getTimestamps(timeInterval))
-      if (isRight(response)) {
-        if (response.value.data.error) throw new Error(response.value.data.error)
-
-        const tickers = response.value.data.tickers
-        if (tickers.length === 0) return null
-
-        const validCurrency = currency === ptTicker ? supportedCurrencies.USD : currency ?? supportedCurrencies.USD
-
-        const initialPrice = tickers[0].prices[validCurrency]
-        const records = tickers
-          .map((ticker) => {
-            const value = ticker.prices[validCurrency]
-            if (value === undefined) return undefined
-            const {changePercent, changeValue} = priceChange(initialPrice, value)
-            const label = new Date(ticker.timestamp).toLocaleString(languageCode, {
-              dateStyle: 'short',
-              timeStyle: 'short',
-            })
-            return {label, value, changePercent, changeValue}
+      // force queryFn to be async, otherwise it takes longer and doesn't show isFetching
+      await delay(0)
+
+      const tickers = ptPriceQuery?.data ?? []
+      if (tickers.length === 0) throw new Error('No PT price data')
```

Comment:
```suggestion
      if (tickers.length === 0) return []
```

Reply:
We're showing the placeholder when fetching is done and data is undefined
```
          {!data ? <ChartPlaceholder /> : <TokenChart onValueSelected={handleChartSelected} dataSources={data} />}
```


If we returned an empty array, it would be used as an actual chart with no points.